### ISSUE TITLE
CI ROS test timeout

### ISSUE THREAD

**Issue by mcsauder** [PX4/Firmware#11290]
**Describe the bug**
The CI ROS tests frequently timeout.  Sometimes after a rebase a previously passing PR will fail, and other times after a rebase against current master the same tests will pass.

![image](https://user-images.githubusercontent.com/2497951/51648957-e2347480-1f3f-11e9-9bdb-5e6033131635.png)

At the time of this writing, the following PR(s) exhibit this failure mode:
http://ci.px4.io:8080/blue/organizations/jenkins/PX4_misc%2FFirmware-SITL_tests/detail/PR-11194/20/pipeline
http://ci.px4.io:8080/blue/organizations/jenkins/PX4_misc%2FFirmware-SITL_tests/detail/PR-11260/11/pipeline

**To Reproduce**
Rebase a current PR against master and force push to remote.  This will re-run/restart the CI tests, and sometimes everything will run smoothly, however, sometimes the tests will timeout.

**Expected behavior**
@dagar, if a timeout occurs, is there a way to automatically restart the test, (perhaps a finite number of times), or allow users to manually restart failing tests through the web interface?

**Comment by dagar** [comment]
Duplicate of https://github.com/PX4/Firmware/issues/11257

**Comment by lamping7** [comment]
This isn't a load-based timeout. There is a failure of the software env not monitored by the test itself.

**Comment by mcsauder** [comment]
Another occurrence of the issue reported here: http://ci.px4.io:8080/blue/organizations/jenkins/PX4_misc%2FFirmware-SITL_tests/detail/PR-11314/1/pipeline

(I don't believe that the LOC in this PR could impact the ROS tests.)

**Comment by mcsauder** [comment]
PR #11314 is passing now after a rebase with current PX4/Firmware master.

**Comment by mcsauder** [comment]
PR 11189 currently shows this behavior: http://ci.px4.io:8080/blue/organizations/jenkins/PX4_misc%2FFirmware-SITL_tests/detail/PR-11189/1/pipeline

**Comment by lamping7** [comment]
#11308 and #11189 are different from the original issue you posted about. Those are preflight fails for accels.

**Comment by mcsauder** [comment]
Thanks @lamping7 , this is what I see at the bottom of #11308:

```[Testcase: testVTOL_mission_1] ... ERROR!
ERROR: max time [300.0s] allotted for test [VTOL_mission_1] of type [px4/mission_test.py]
```


But you're right about #11189, 

```
WARN  [commander] Preflight Fail: Accels inconsistent - Check Cal


Exiting...

```

I'll delete the reference to #11189, I think I autocompleted an incorrect PR.  I'll look again.  Thanks!

**Comment by lamping7** [comment]
Scroll up on #11308. The accel problem is not associated with these PRs. It seems something crept into master.

**Comment by mcsauder** [comment]
@lamping, #11305 still shows the same behaviors, so at least that instance is still relevant.  I watched #11189 automagically reload the page about 20 times and keep scrolling my page to the bottom shortly after your post, so I honestly can't say if it was the same at the point you saw it as when I saw it... if the CI tools run again it definitely yields different results, I've seen this many times in the past few months with rebasing my own PRs.  Any ideas?  How can I help out?
